unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,334 @@
+    """Include tests for the constructor of the RedisAggregator class."""
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_grid_fee_calculation_instantiated():
+        grid_fee_calculation_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.GridFeeCalculation",
+                   return_value=grid_fee_calculation_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+            assert agg.grid_fee_calculation is grid_fee_calculation_mock
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_redis_db_instantiated():
+        redis_db_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.StrictRedis.from_url",
+                   return_value=redis_db_mock) as redis_from_url:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            redis_from_url.assert_called_with(LOCAL_REDIS_URL)
+            assert agg.redis_db is redis_db_mock
+
+    @staticmethod
+    def test_constructor_pubsub_instantiated(aggregator):
+        aggregator.redis_db.pubsub.assert_called()
+        assert aggregator.pubsub is aggregator.redis_db.pubsub()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_client_command_buffer_instantiated():
+        client_command_buffer_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer",
+                   return_value=client_command_buffer_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+        assert agg._client_command_buffer is client_command_buffer_mock
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_threadpoolexecutor_instantiated():
+        ThreadPoolExecutor_mock = MagicMock
+        with patch("gsy_e_sdk.redis_aggregator.ThreadPoolExecutor",
+                   return_value=ThreadPoolExecutor_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+        assert agg.executor is ThreadPoolExecutor_mock
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_connect_and_subscribe_side_effects_1():
+        """Test  side effects due to the methods called with _connect_and_subscribe().
+        -> Side effects of _subscribe_to_aggregator_response_and_start_redis_thread
+        """
+
+        aggregator = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+        channel_dict_1 = {"aggregator_response": aggregator._aggregator_response_callback}
+
+        aggregator.pubsub.run_in_thread.assert_called_with(daemon=True)
```

Comment:
Can you please explain? How can the `aggregator.pubsub.run_in_thread` be a mock? I do not see the patch or mock command.

Reply:
The same applies to the `aggregator.redis_db.publish` in the next test method.

Reply:
I see it in multiple methods here and wonder how this works. Thanks a lot in advance for explaining!

Reply:
@hannesdiedrich the reason is that `StrictRedis` and `ThreadPoolExecutor` have been mocked in this fixture: https://github.com/gridsingularity/gsy-e-sdk/pull/197/files#diff-701f7f3f361fabc75b8c82fd7b3833b77782af10ec9c24b196a4c6c9524de42eR58-R63
The fixture has `autouse=True`, which means that it is automatically applied to all individual test functions in this module.

I'm usually a bit skeptical of using `autouse=True` because it's a bit too broad for my personal taste (it could be used in functions that don't need it). But I suppose that in this case Eduardo wanted to avoid mistakenly calling the "real" redis implementation. That's why in a previous comment I suggested using [fakeredis](https://github.com/jamesls/fakeredis) instead, to deal with this in all tests.

Reply:
What Pierpaolo explains is precisely what happens in these tests. 
In the fixture he mentions, the StrictRedis class and ThreadPoolExecutor are mocked to avoid those external connections. 
I set `autouse=True` for them to be used everytime an aggregator instance gets created (which applies to each test case in this case). And therefore, for each instance of the aggregator, this attributes would be mocked and therefore no connections would be stablished. 

I had not understand your previous suggestion about the fakeredis. Regarding that...
If we follow this approach, should I set a patch for StrictRedis whose return value is the fakeredis instance? Or how should that be handled?

Reply:
I was hoping that fakeredis would work like fakes3 (where you just point to a different URL to use the fake server). Unfortunately, it looks like you actually need to import it directly and use it to patch the "real" redis classes (similar to what you've done here). So, in my opinion, it's not as nice as I originally thought.
If you have the time, you could check the possible alternatives and tell us if you find something that you think is worth it. But I wouldn't consider this to be part of this PR.

Reply:
I'll have a look and see if some nice refactor can be achieved by using fakeredis.